### ISSUE TITLE
Consider adding ApplicationStatus health check to ServiceDefaults in place of default "live" check

### ISSUE THREAD

**Issue by DamianEdwards** [dotnet/aspire#4110]
The ServiceDefaults templates today add a default "live" health check that simply returns `true` as a way of ensuring the application is running and responsive. We should consider replacing this with the [`AspNetCore.HealthChecks.ApplicationStatus`](https://www.nuget.org/packages/AspNetCore.HealthChecks.ApplicationStatus#readme-body-tab) health check instead, as it bases the status off of the application's `IHostApplicationLifetime` instance, which will change to "unhealthy" as soon as the app starts shutting down.

**Comment by willibrandon** [comment]
Am I correct that a request must be made internally to add this non-MSFT package to a feed?

**Comment by DamianEdwards** [comment]
> Am I correct that a request must be made internally to add this non-MSFT package to a feed?

Yes, but you can modify the `nuget.config` file in the repo root to verify the changes while making the edits.

**Comment by willibrandon** [comment]
It would be cool if health check status was displayed in the dashboard, which I see is being considered in #889.

**Comment by wertzui** [comment]
Wouldn't it be better to have multiple endpoints, like `/health/startup`, `/health/live`, `/health/ready` to correspond to the default Kubernetes probes instead of having one endpoint which reports the status in the body? It might also be easier for other health check tools, like Application Insights, which by default just check for a 2xx status code to report a resource as healthy.

**Comment by martincostello** [comment]
Exposing the three endpoints that k8s expects by default would definitely be useful.

**Comment by DamianEdwards** [comment]
@wertzui @martincostello that's a fair suggestion but a bit separate to this issue. Feel free to log an issue suggesting we move to three standard health endpoints and which health checks they'd use.

**Comment by martincostello** [comment]
> Feel free to log an issue

#5420

**Comment by DamianEdwards** [comment]
Would accept a PR for this but not important enough to put in a milestone for now IMO.

**Comment by mbrown379** [comment]
Hi @DamianEdwards , I added a [PR](https://github.com/dotnet/aspire/pull/7151) to close this issue.
